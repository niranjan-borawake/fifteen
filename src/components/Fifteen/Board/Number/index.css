.number {
  background: var(--br-50-color);
  border: 1px solid var(--br-900-color);
  border-radius: 50px;
  cursor: default;
  display: grid;
  justify-items: center;
  align-items: center;
  color: var(--br-900-color);
  user-select: none;
}
.number.selected {
  background: var(--br-700-color);
  color: var(--br-50-color);
}
.number.used {
  background: var(--br-300-color);
  color: var(--br-900-color);
}
