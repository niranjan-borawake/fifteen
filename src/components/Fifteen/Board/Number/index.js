import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import { ACTIONS } from '../../reducer.js';

import './index.css';

const Number = ({
  number: { value, id, isUsed },
  currentSelections,
  isGameOver,
  dispatch,
}) => {
  const onNumberClick = () => {
    if (isUsed || isGameOver) {
      return;
    }

    if (currentSelections.includes(id)) {
      dispatch({
        type: ACTIONS.REMOVE_FROM_CURRENT_SELECTIONS,
        toBeRemoved: id,
      });
      return;
    }

    if (currentSelections.length < 3) {
      dispatch({
        type: ACTIONS.ADD_TO_CURRENT_SELECTIONS,
        toBeAdded: id,
      });
    }
  };
  return (
    <div
      data-testid={`number-${id}`}
      className={classNames({
        number: true,
        selected: currentSelections.includes(id),
        used: isUsed,
      })}
      onClick={onNumberClick}
    >
      {value}
    </div>
  );
};

Number.propTypes = {
  number: PropTypes.shape({
    value: PropTypes.number.isRequired,
    id: PropTypes.number.isRequired,
    isUsed: PropTypes.bool.isRequired,
  }).isRequired,
  currentSelections: PropTypes.arrayOf(PropTypes.number).isRequired,
  isGameOver: PropTypes.bool.isRequired,
  dispatch: PropTypes.func.isRequired,
};

export default React.memo(Number);
