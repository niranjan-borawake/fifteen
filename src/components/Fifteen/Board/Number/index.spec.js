import { render } from '@testing-library/react';
import Number from '.';

describe('Number', () => {
  it('should render a given number', () => {
    expect(
      render(
        <Number
          dispatch={jest.fn}
          number={{ value: 5, id: 1, isUsed: false }}
          isGameOver={false}
          currentSelections={[0]}
          isBotTurn={false}
        />
      ).container
    ).toMatchSnapshot();
  });
});
