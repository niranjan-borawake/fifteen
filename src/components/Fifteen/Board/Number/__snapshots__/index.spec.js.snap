// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Number should render a given number 1`] = `
<div>
  <div
    class="number"
    data-testid="number-1"
  >
    5
  </div>
</div>
`;
